src/Agent.Worker/Release/ReleaseFileSystemManager.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    [ServiceLocator(Default = typeof(ReleaseFileSystemManager))]
+    public interface IReleaseFileSystemManager : IAgentService
+    {
+        IEnumerable<FileInfo> GetFiles(string directoryPath, SearchOption searchOption);
+
+        StreamReader GetFileReader(string filePath);
+
+        Task WriteStreamToFile(Stream stream, string filePath);
+
+        void DeleteDirectory(string directoryPath);
+
+        void CleanupDirectory(string directoryPath);
+    }
+
+    public class ReleaseFileSystemManager : AgentService, IReleaseFileSystemManager
+    {
+        private RetryExecutor _retryExecutor = new RetryExecutor();
+        private const int StreamBufferSize = 1024;
+
+        public IEnumerable<FileInfo> GetFiles(string directoryPath, SearchOption searchOption)
+        {
+            return
+                Directory.GetFiles(ValidatePath(directoryPath), "*", searchOption)
+                    .Select(fullPath => new FileInfo(fullPath));
+        }
+
+        public IEnumerable<string> GetDirectories(string directoryPath, SearchOption searchOption)
+        {
+            return Directory.GetDirectories(ValidatePath(directoryPath), "*", searchOption);
+        }
+
+        public bool FileExists(string filePath)
+        {
+            return File.Exists(filePath);
+        }
+
+        public void DeleteFile(string filePath)
+        {
+            _retryExecutor.Execute(File.Delete, filePath);
```

Comment:
This function is not required, I moved the entire helper file from old code and it was left over.. removed the unused code now..